plugins/lsp/schemastore.nix

```diff
@@ -0,0 +1,151 @@
+{
+  lib,
+  helpers,
+  config,
+  pkgs,
+  ...
+}:
+with lib;
+helpers.vim-plugin.mkVimPlugin config {
+  name = "schemastore";
+  originalName = "SchemaStore.nvim";
+  defaultPackage = pkgs.vimPlugins.SchemaStore-nvim;
+
+  maintainers = [ maintainers.GaetanLepage ];
+
+  extraOptions =
+    let
+      schemaOpts =
+        let
+          schemaEntry = types.submodule {
```

Comment:
Should this be freeformType as it corresponds to plugin settings?

Reply:
I have added `freeformType`.

Reply:
I'm still conflicted with submodules; on the one hand we're specifying some options can't be null and on the other hand we're saying it's a freeform type :joy: 

I guess freeform only really affects undeclared options?

Reply:
Yes, exactly. If you explicitly declare an option, it will be constrained in terms of type.
The `freeformType` allows you to provide values for arbitrary undeclared options.
They are not at all contradictory.